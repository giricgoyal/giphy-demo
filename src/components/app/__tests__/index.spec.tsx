import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import App from '../index'
import { setupIntersectionObserverMock } from '../../../mocks'

describe('<App />', () => {
    let container
    beforeEach(() => {
        setupIntersectionObserverMock()
        ;({ container } = render(<App />))
    })

    test('should match snapshot', () => {
        expect(container).toMatchSnapshot()
    })
})
