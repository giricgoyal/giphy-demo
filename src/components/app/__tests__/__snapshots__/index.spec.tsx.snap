// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div>
  <div
    class="page container"
  >
    <nav
      class="navbar navbar-expand navbar-light"
      data-testid="navbar"
    >
      <div
        class="container"
      >
        <a
          class="navbar-brand"
          href="#"
        >
          GIPHY-DEMO
        </a>
      </div>
    </nav>
    <div
      class="giphy-home container"
      data-testid="giphy-home"
    >
      <form
        class="search"
      >
        <input
          class="search__input form-control"
          data-testid="input"
          placeholder="Search"
          type="text"
          value=""
        />
        <svg
          class="search__icon icon"
        >
          <use
            xlink:href="#undefined"
          />
        </svg>
        
      </form>
      <div
        class="grid"
        data-testid="grid"
      >
        <div
          class="grid__header"
        >
          <svg
            class="undefined icon"
          >
            <use
              xlink:href="#undefined"
            />
          </svg>
          <span>
            Trending
          </span>
        </div>
        <div
          class="row"
        >
          <div
            class="col"
          />
          <div
            class="col"
          />
          <div
            class="col"
          />
          <div
            class="col"
          />
        </div>
        <div
          class="grid__loader"
        >
          <div
            class="spinner-border"
          />
        </div>
        <div />
      </div>
    </div>
  </div>
</div>
`;
